### PR TITLE
Make unified register PointerObjects share RegisterNode in the PointsToGraph

### PR DESCRIPTION
**Pull request by haved**
Since `RegisterNode`s support mapping multiple RVSDG outputs to a single `PointsToGraph` node, when they all share pointees, it makes sense to use this for unified `PointerObject`s of Register kind.

This PR also adds a test of `Andersen::ConstructPointsToGraphFromPointerObjectSet`, which was missing.

Just for fun, here is the PointsToGraph being built in the test:

ptg

### CODE REVIEW

File: jlm/llvm/opt/alias-analyses/Andersen.cpp [hunk 1]
```diff
@@ -899,7 +899,7 @@ Andersen::Analyze(const RvsdgModule & module)
 std::unique_ptr<PointsToGraph>
 Andersen::ConstructPointsToGraphFromPointerObjectSet(
     const PointerObjectSet & set,
-    Statistics & statistics)
+    Statistics * statistics)
```
Comment (phate) [inline jlm/llvm/opt/alias-analyses/Andersen.cpp:902]: Why did you change this from a reference to a pointer?

Comment (phate) [reply jlm/llvm/opt/alias-analyses/Andersen.cpp:902]: Is it because of the unit test? Could you not just create one there?

**Review by phate — APPROVED** [review]
Only a minor issue.

File: jlm/llvm/opt/alias-analyses/Andersen.cpp (hunk 1 above)
Comment (haved) [reply jlm/llvm/opt/alias-analyses/Andersen.cpp:902]: That is what I started doing, but I realized the Statistics class is only forward declared in the header, while the definition is in `Andersen.cpp`.   Without moving it, or declaring a public `Andersen:: CreateStatistics()`, I can't make instances of Statistics from the test.

I don't know if it is possible to forward declare `Andersen::Statistics::Create()`, but googling doesn't give me any examples of it being allowed.

Comment (phate) [reply jlm/llvm/opt/alias-analyses/Andersen.cpp:902]: Could you provide a `ConstructPointsToGraphFromPointerObjectSet()` method that does not take a `Statistics` and internally is defined in the `cpp` file and calls the method version with `Statistics` parameter internally?

Comment (haved) [reply jlm/llvm/opt/alias-analyses/Andersen.cpp:902]: Good point, that works

File: jlm/llvm/opt/alias-analyses/Andersen.cpp [hunk 2]
```diff
@@ -1003,4 +1006,12 @@ Andersen::ConstructPointsToGraphFromPointerObjectSet(
   return pointsToGraph;
 }
 
+std::unique_ptr<PointsToGraph>
+Andersen::ConstructPointsToGraphFromPointerObjectSet(const PointerObjectSet & set)
+{
+  // Create a throwaway instance of statistics
+  auto statistics = Statistics::Create(jlm::util::filepath(""));
```
Comment (phate) [inline jlm/llvm/opt/alias-analyses/Andersen.cpp:1013]: There is no need to create an instance on the heap here. It could be on the stack, no?

**Review by phate — DISMISSED** [review]
I would rather create the statistics on the stack instead of the heap here.

**Review by phate — APPROVED** [review]
